### PR TITLE
Nodejs week1 hiren

### PR DESCRIPTION
**Pull request by hirenkumar91**
Home work week 1 Node JS 

future-meals | Respond with all meals in the future (relative to the when datetime)
past-meals | Respond with all meals in the past (relative to the when datetime)
all-meals | Respond with all meals sorted by ID
first-meal | Respond with the first meal (meaning with the minimum id)
last-meal | Respond with the last meal (meaning with the maximum id)

### CODE REVIEW

File: api/src/routers/all-meal.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const allMealroute = express.Router();
+
+allMealroute.get("/", async (req, res, next) => {
+    console.log("Received request for /api/all-meal");
```
Comment (urbanogilson) [inline api/src/routers/all-meal.js]: I recommend taking a look at how to use a debugger, it's an interesting and super powerful tool

https://code.visualstudio.com/docs/nodejs/nodejs-debugging

```suggestion
```

File: api/src/routers/all-meal.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    try{
+        const all_meals = await knex("meal").orderBy("id");
+    res.json(all_meals);
+    } catch (err) {
+        next(err);
+      }
```
Comment (urbanogilson) [inline api/src/routers/all-meal.js]: Please user Prettier to format the code

File: api/src/routers/first-meal.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const firstMealroute = express.Router();
+
+firstMealroute.get("/", async (req, res, next) => {
+    console.log("Received request for /api/first-meal");
+    try{
+        const first_meals = await knex("meal").where("id",">",0).orderBy("id","asc").first();
```
Comment (urbanogilson) [inline api/src/routers/first-meal.js]: In JS, the convention is camelCase https://www.freecodecamp.org/news/snake-case-vs-camel-case-vs-pascal-case-vs-kebab-case-whats-the-difference/

ID's are usually non-negative and auto incremental, so the `.where("id",">",0).` is not required

The last suggestion is to not make explicit the implicit values. The orderBy is `asc` by default

```suggestion
        const firstMeal = await knex("meal").orderBy("id").first();
```

File: api/src/routers/future-meal.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const futureMealroute = express.Router();
+
+futureMealroute.get("/", async (req, res, next) => {
+    console.log("Received request for /api/all-meal");
+    try{
+        const futureMeal_meals = await knex("meal").where("when",">",new Date()).orderBy("id");
```
Comment (urbanogilson) [inline api/src/routers/future-meal.js]: The Date time field requires a special representation in SQL. You can explore knex helpers like https://knexjs.org/guide/utility.html#now to solve these common issues

```suggestion
        const futureMeal_meals = await knex("meal").where("when",">",knex.fn.now()).orderBy("id");
```

**Review by urbanogilson — COMMENTED** [review]
Good job! Left inline comments

File: api/src/routers/all-meal.js (hunk 1 above)
Comment (hirenkumar91) [reply api/src/routers/all-meal.js]: thanks for Recomandation will be using it

File: api/src/routers/first-meal.js (hunk 3 above)
Comment (hirenkumar91) [reply api/src/routers/first-meal.js]: Thank you for Suggestions made changes as told.

File: api/src/routers/all-meal.js (hunk 2 above)
Comment (hirenkumar91) [reply api/src/routers/all-meal.js]: using it now and updated all codes

File: api/src/routers/future-meal.js (hunk 4 above)
Comment (hirenkumar91) [reply api/src/routers/future-meal.js]: thank you for feeedback , made changes accordingly

**Comment by hirenkumar91** [comment]
Dear Mentor , i have made all changes as told should i merge

File: api/src/routers/last-meal.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const lastMealroute = express.Router();
+
+lastMealroute.get("/", async (req, res, next) => {
+    console.log("Received request for /api/last-meal");
+    try{
+        const last_meals = await knex("meal").where("id",">",0).orderBy("id","desc").first();
```
Comment (urbanogilson) [inline api/src/routers/last-meal.js:9]: ID is incremental starting from 1so this query is not required
```suggestion
        const last_meals = await knex("meal").orderBy("id","desc").first();
```

File: api/src/routers/past-meal.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const pastMealroute = express.Router();
+
+pastMealroute.get("/", async (req, res, next) => {
+    console.log("Received request for /api/all-meal");
```
Comment (urbanogilson) [inline api/src/routers/past-meal.js:7]:
```suggestion

```

File: api/src/routers/last-meal.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const lastMealroute = express.Router();
+
+lastMealroute.get("/", async (req, res, next) => {
+    console.log("Received request for /api/last-meal");
```
Comment (urbanogilson) [inline api/src/routers/last-meal.js:7]:
```suggestion
```

File: api/src/routers/future-meal.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const futureMealroute = express.Router();
+
+futureMealroute.get("/", async (req, res, next) => {
+    console.log("Received request for /api/all-meal");
```
Comment (urbanogilson) [inline api/src/routers/future-meal.js:7]:
```suggestion
```

File: api/src/routers/first-meal.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const firstMealroute = express.Router();
+
+firstMealroute.get("/", async (req, res, next) => {
+  console.log("Received request for /api/first-meal");
```
Comment (urbanogilson) [inline api/src/routers/first-meal.js:7]:
```suggestion
```

File: api/src/index.js [hunk 10]
```diff
@@ -2,30 +2,33 @@ import "dotenv/config";
...
+apiRouter.use("/all-meal", allMealroute);
+apiRouter.use("/future-Meal", futureMealroute);
+apiRouter.use("/past-Meal", pastMealroute);
+apiRouter.use("/first-meal", firstMealroute);
+apiRouter.use("/last-meal", lastMealroute);
 app.use("/api", apiRouter);
-
-app.listen(process.env.PORT, () => {
+const PORT = process.env.PORT || 3000;
+app.listen(PORT, () => {
```
Comment (urbanogilson) [inline api/src/index.js:32]:
```suggestion

app.listen(process.env.PORT || 3000, () => {
```

**Pull request merged**